trunner/tools/phoenix.py

```diff
@@ -119,8 +120,10 @@ def _run(self):
             # such wait avoids a following issue with loading an app:
             # https://github.com/phoenix-rtos/phoenix-rtos-project/issues/545
             time.sleep(0.5)
-        except (TimeoutError, Exception) as exc:
-            raise PhoenixdError(str(exc)) from exc
+        except (TimeoutError, AssertionError) as exc:
+            raise PhoenixdError(msg=str(exc)) from exc
+        except Exception as exc:
+            raise PhoenixdError(msg=str(exc), output=traceback.format_exc()) from exc
```

Comment:
hmm, it looks strange that the traceback is under `OUTPUT`, imho would be better to format the `PhoenixdError` exception directly inside the the exception implementation, sample code below (not tested):
```python
class ProcessError(HarnessError):
    name: str = "PROCESS"
    # [...]

    def __str__(self) -> str:
        # TODO format github actions output
        err = [bold(f"{self.name} ERROR: ") + self.msg]

        if self.output is not None:
            err.extend([bold("OUTPUT:"), self.output])

        err.extend(self._format_additional_info())

        # ADDED CODE:
        # add traceback if it's rather an implementation error instead of a logic one
        if not isinstance(self.__cause__, TimeoutError):
            err.append(bold("TRACEBACK:"))
            err.extend(traceback.format_exception(self))

        err.append("")
        return "\n".join(err)
```

Reply:
I've looked through all ProcessErrors used in test runner and in most of the cases traceback seems to be unnecessary (the line in test runner is then shown), I'd like to print it only for internal exceptions, where we don't know the exact line. That's why I've proposed adding it only when `Exception` is raised and labeled as `TRACEBACK`:

![Screenshot from 2023-11-02 13-42-59](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/77304431/9772a71a-a666-4196-b99b-7dbb2efef854)

What do you think?